data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3ObjectRequest.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.source;
+
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.plugins.source.codec.Codec;
+import org.opensearch.dataprepper.plugins.source.compression.CompressionEngine;
+import org.opensearch.dataprepper.plugins.source.configuration.S3SelectSerializationFormatOption;
+import org.opensearch.dataprepper.plugins.source.ownership.BucketOwnerProvider;
+import software.amazon.awssdk.services.s3.S3AsyncClient;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.CompressionType;
+import software.amazon.awssdk.services.s3.model.FileHeaderInfo;
+
+import java.time.Duration;
+import java.util.function.BiConsumer;
+public class S3ObjectRequest {
```

Comment:
This class should have a unit test.

Reply:
Thanks David for your review, Changes are incorporated and available for review.

Reply:
Thanks David for your review, Changes are incorporated and available for review.